### PR TITLE
Feat: add filters in hardware details

### PR DESCRIPTION
**Pull request by Francisco2002**
This PR refactor de logic of filters and add build, boots and tests duration filters.

Like duration in hardware details are nullable, I am sending a patch test to allow the test of duration filters

### CODE REVIEW

File: backend/kernelCI_app/views/hardwareDetailsView.py [hunk 1]
```diff
@@ -153,79 +156,221 @@ class HardwareDetails(View):
...
+
+    def __handle_boot_duration(self, current_filter):
+        value = current_filter["value"][0]
+        operation = current_filter["comparison_op"]
+        if operation == "lte":
+            self.filterBootDurationMax = toIntOrDefault(value, None)
+        else:
+            self.filterBootDurationMin = toIntOrDefault(value, None)
+
+    def __handle_test_status(self, current_filter):
```
Comment (WilsonNet) [inline backend/kernelCI_app/views/hardwareDetailsView.py:196]: don't use `__` that was a missconception that we had, private methods have only one underscore before them

also, these handlers are very close to the one in hardware details view, se if you can unite them

the trick is to receive what they are modifying as a parameter
or to return the value
I'd rather have them returning the value

File: backend/kernelCI_app/utils.py [hunk 2]
```diff
@@ -2,9 +2,11 @@
 from typing import Union, TypedDict, List, Optional, Dict
 from django.utils import timezone
 from datetime import timedelta
+from django.http import HttpResponseBadRequest
 import re
 
 DEFAULT_QUERY_TIME_INTERVAL = {"days": 7}
+NULL_STRINGS = ["null", "Unknown"]
```
Comment (murilx) [inline backend/kernelCI_app/utils.py:9]: You should use a set in this case

File: backend/kernelCI_app/views/hardwareDetailsView.py [hunk 3]
```diff
@@ -23,7 +25,7 @@
 NULL_STRINGS = ["null", "Unknown"]
```
Comment (murilx) [inline backend/kernelCI_app/views/hardwareDetailsView.py:25]: You can import this from utils

File: backend/kernelCI_app/views/hardwareDetailsView.py [hunk 4]
```diff
@@ -153,79 +156,221 @@ class HardwareDetails(View):
...
+            not record['build__architecture']
+            and any(null_string in self.filterArchitecture for null_string in NULL_STRINGS)
+        )
+        record_compiler_null_accept = (
+            not record['build__compiler']
+            and any(null_string in self.filterTreeDetailsCompiler for null_string in NULL_STRINGS)
+        )
+        record_config_null_accept = (
+            not record['build__config_name']
+            and any(null_string in self.filterTreeDetailsConfigs for null_string in NULL_STRINGS)
```
Comment (murilx) [inline backend/kernelCI_app/views/hardwareDetailsView.py:254]: Is there a reason here to not just use `self.filterArchitecture in NULL_STRINGS` (the same applies to the other instances of `any(null_string in _ for null_string in NULL_STRINGS)`?

Comment (Francisco2002) [reply backend/kernelCI_app/views/hardwareDetailsView.py:254]: `self.filterArchitecture` is a set of architecture to filter.
`self.filterTreeDetailsCompiler` and `filterTreeDetailsConfigs` too

File: backend/kernelCI_app/views/treeDetailsView.py [hunk 5]
```diff
@@ -442,7 +378,8 @@ def get(self, request, commit_hash: str | None):
         origin_param = request.GET.get("origin")
         git_url_param = request.GET.get("git_url")
         git_branch_param = request.GET.get("git_branch")
-        self.__processFilters(request)
+        self.filterParams = FilterParams(request)
+        self.setup_filters()
```
Comment (WilsonNet) [inline backend/kernelCI_app/views/treeDetailsView.py:382]: much better now

File: backend/kernelCI_app/utils.py [hunk 6]
```diff
@@ -111,9 +113,137 @@ class FilterParams:
...
+        try:
+            for current_filter in self.filters:
+                field = current_filter["field"]
+                # Delegate to the appropriate handler based on the field
+                if field in self.filter_handlers:
+                    self.filter_handlers[field](current_filter)
+        except InvalidComparisonOP as e:
+            return HttpResponseBadRequest(getErrorResponseBody(str(e)))
+
+    def create_filters_from_body(self, body: Dict):
```
Comment (WilsonNet) [inline backend/kernelCI_app/utils.py:214]: one day we need to make this a little better but the improvements here are good for today

**Review by WilsonNet — APPROVED** [review]
niceo one

**Review by murilx — APPROVED** [review]
Nicely done!

**Pull request merged**